third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -89,6 +92,7 @@ impl Interpreter {
         args: Vec<Value>,
         data_store: &mut TransactionDataCache,
         module_store: &ModuleStorageAdapter,
+        module_storage: &impl ModuleStorage,
```

Comment:
But we cannot not have both until we upgrade/enable the feature? It will be one, but we will need to wait for a few months before that. In the meantime, can mark V1 as deprecated?

Reply:
I think we can take an enum and dispatch based on that? since v1 and v2 are not used at the same time?
```
enum ModuleStorage {
  V1(..),
  V2(..),
}
```

Reply:
Ah, in that sense! Let me try to do that to clean-up things a bit

Reply:
Does the enum actually make sense? The code paths are quite different, also we sometimes take transaction data cache for V1, I am wondering if this really encapsulates anything?